### PR TITLE
Adding support for v1.25.0 k8s

### PR DESCRIPTION
**Pull request by gizas**
## What does this PR do?

Adds the new version of v1.25.0 in the list of the supported/tested versions.

Part of https://github.com/elastic/observability-dev/issues/2195

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -224,7 +224,7 @@ pipeline {
         axes {
           axis {
             name 'K8S_VERSION'
-            values "v1.24.0", "v1.23.6", "v1.22.9", "v1.21.12"
+            values "v1.25.0","v1.24.3", "v1.23.9", "v1.22.12"
```
Comment (ChrsMark) [inline .ci/Jenkinsfile:227]: Are we sure we want to remove testing for `v1.21` already?

Comment (ChrsMark) [reply .ci/Jenkinsfile:227]: Well it's already EoL https://endoflife.date/kubernetes, let's reduce the list to imrpove CI time

**Review by ChrsMark — APPROVED** [review]

**Comment by gizas** [comment]
Wait until kindest/node:v1.25.0 to be relased. Still in beta
https://hub.docker.com/r/kindest/node/tags?page=1&name=1.25

**Pull request merged**